node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -324,8 +324,7 @@ impl<C: Context> ActiveValidator<C> {
             panorama = state.panorama().clone();
         }
         let seq_number = panorama.next_seq_num(state, self.vidx);
-        // TODO: After LNC we won't always need all known endorsements.
-        let endorsed = state.endorsements().collect();
+        let endorsed = panorama.need_endorsements(state);
```

Comment:
Is it possible to construct an active validator that needs needs endorsements so we can test this functionality?  Then, we could check to make sure the resulting `SignedWireVote` is endorsed.

Reply:
Fortunately, all the `active_validator`'s state does here is irrelevant to the endorsements logic. That depends only on the input arguments to that function.